torchvision/models/regnet.py

```diff
@@ -0,0 +1,785 @@
+
+        The second step is to compute the parameters per stage,
+        taking into account the skip connection and the final 1x1 convolutions.
+        We use the fact that the output width is constant within a stage.
+        """
+
+        QUANT = 8
+        STRIDE = 2
+
+        # Compute the block widths. Each stage has one unique block width
+        widths_cont = np.arange(self.depth) * self.w_a + self.w_0
+        block_capacity = np.round(np.log(widths_cont / self.w_0) / np.log(self.w_m))
+        block_widths = (
+            np.round(np.divide(self.w_0 * np.power(self.w_m, block_capacity), QUANT))
+            * QUANT
+        )
+        num_stages = len(np.unique(block_widths))
+        block_widths = block_widths.astype(int).tolist()
+
+        # Convert to per stage parameters
+        split_helper = zip(
+            block_widths + [0],
+            [0] + block_widths,
+            block_widths + [0],
+            [0] + block_widths,
+        )
+        splits = [w != wp or r != rp for w, wp, r, rp in split_helper]
+
+        stage_widths = [w for w, t in zip(block_widths, splits[:-1]) if t]
+        stage_depths = np.diff([d for d, t in enumerate(splits) if t]).tolist()
+
+        strides = [STRIDE] * num_stages
+        bottleneck_multipliers = [self.bottleneck_multiplier] * num_stages
+        group_widths = [self.group_width] * num_stages
+
+        # Adjust the compatibility of stage widths and group widths
+        stage_widths, group_widths = self._adjust_widths_groups_compatibilty(
+            stage_widths, bottleneck_multipliers, group_widths
+        )
+
+        return zip(
+            stage_widths, strides, stage_depths, group_widths, bottleneck_multipliers
+        )
+
+    @staticmethod
+    def _adjust_widths_groups_compatibilty(
+            stage_widths: List[int], bottleneck_ratios: List[float],
+            group_widths: List[int]) -> Tuple[List[int], List[int]]:
+        """
+        Adjusts the compatibility of widths and groups,
+        depending on the bottleneck ratio.
+        """
+        # Compute all widths for the current settings
+        widths = [int(w * b) for w, b in zip(stage_widths, bottleneck_ratios)]
+        group_widths_min = [min(g, w_bot) for g, w_bot in zip(group_widths, widths)]
+
+        # Compute the adjusted widths so that stage and group widths fit
+        ws_bot = [_make_divisible(w_bot, g) for w_bot, g in zip(widths, group_widths_min)]
+        stage_widths = [int(w_bot / b) for w_bot, b in zip(ws_bot, bottleneck_ratios)]
+        return stage_widths, group_widths_min
+
+
+class RegNet(nn.Module):
+    def __init__(self, params: RegNetParams) -> None:
+        super().__init__()
+
+        activation = params.activation(inplace=True)
+
+        # Ad hoc stem
+        self.stem = params.stem_type(
+            3,  # width_in
+            params.stem_width,
+            params.bn_epsilon,
+            params.bn_momentum,
+            activation,
+        )
```

Comment:
I think it is a bit more generic to let the user pass the fully-materialized `stem` in the constructor, so that we don't get fixed on the particular arguments that are necessary to construct the `stem`